zap/src/test/java/org/parosproxy/paros/network/HttpHeaderUnitTest.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2024 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.parosproxy.paros.network;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+
+import org.apache.commons.httpclient.URI;
+import org.apache.commons.httpclient.URIException;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+class HttpHeaderUnitTest {
+
+    @ParameterizedTest
+    @ValueSource(
+            strings = {
+                "application/taxii+json",
+                "application/vnd.api+json",
+                "text/json",
+                "application/yaml",
+                "application/x-yaml",
+                "text/yaml",
+                "application/xml",
+                "application/problem+xml",
+                "text/xml",
+                "image/webp",
+                "image/gif",
+                "image/jpeg"
+            })
+    void shouldIdentifyContentTypes(String type)
+            throws URIException, HttpMalformedHeaderException, NullPointerException {
+        // Given
+        HttpMessage msg = new HttpMessage(new URI("https://example.com/", true));
+        msg.getResponseHeader().setHeader(HttpHeader.CONTENT_TYPE, type);
+        String[] acceptedTypes = {"json", "xml", "yaml", "image"};
+        // When
+        boolean hasType = msg.getResponseHeader().hasContentType(acceptedTypes);
```

Comment:
Should really test the `HttpHeader` not one of the implementations.

Reply:
You mean create just a header not a whole message?

Reply:
Right, it should just need the `HttpHeader`.